### PR TITLE
Run kuttl tests job for designate-operator

### PR DESCRIPTION
**Pull request by bshewale**
Depends-On: https://github.com/openstack-k8s-operators/designate-operator/pull/91

### CODE REVIEW

**Comment by bshewale** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy-kuttl

File: ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml [hunk 1]
```diff
@@ -29,6 +66,17 @@ tests:
...
   container:
     from: openstack-operator-ci-pre-commit
+- as: designate-operator-build-deploy-kuttl
+  cluster_claim:
+    architecture: amd64
+    cloud: vexxhost
+    owner: openstack-k8s-operators
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.10"
```
Comment (cescgina) [inline ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:76]: I think that the failure is not random and there are no more 4.10 sno clusters available? This should maybe be `4.12` as in https://github.com/openshift/release/blob/90a763ca619811e5f847db4b9c6010dc35706643/ci-operator/config/openstack-k8s-operators/openstack-operator/openstack-k8s-operators-openstack-operator-main.yaml#L83? @Sandeepyadav93 could you clarify?

Comment (Sandeepyadav93) [reply ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:76]: correct, we moved to 4.12

Comment (cescgina) [reply ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:76]: @bshewale please update the version in line 76 as well

Comment (bshewale) [reply ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml]: Done

**Comment by cescgina** [comment]
@bshewale the failure is a legit issue with the designate kuttl tests, this PR should fix it: https://github.com/openstack-k8s-operators/designate-operator/pull/91

**Comment by bshewale** [comment]
> @bshewale the failure is a legit issue with the designate kuttl tests, this PR should fix it: [openstack-k8s-operators/designate-operator#91](https://github.com/openstack-k8s-operators/designate-operator/pull/91)

Thanks @cescgina :) Will edit commit message with Depends-On tag.

**Comment by cescgina** [comment]
> > @bshewale the failure is a legit issue with the designate kuttl tests, this PR should fix it: [openstack-k8s-operators/designate-operator#91](https://github.com/openstack-k8s-operators/designate-operator/pull/91)
> 
> Thanks @cescgina :) Will edit commit message with Depends-On tag.

I don't think depends-on works here, we'll need to merge the designate PR first

**Comment by cescgina** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy-kuttl

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+    - ^main-
+    cluster: build03
+    context: ci/prow/designate-operator-build-deploy-kuttl
+    decorate: true
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy-kuttl
+    optional: true
+    rerun_command: /test designate-operator-build-deploy-kuttl
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:16]: Add:
`skip_if_only_changed: ^(renovate.json|.*\.md|tests/functional/.*|OWNERS.*|LICENSE.*)$`
after `rerun_command` key

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 3]
```diff
@@ -1,5 +1,67 @@
 presubmits:
   openstack-k8s-operators/designate-operator:
+  - agent: kubernetes
+    always_run: true
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:4]: Set this to `false`
Together with skip_if_only_changed mentioned below, this job will always run and skip when PR only have irrelevant fles

**Review by viroel — COMMENTED** [review]
Hi @bshewale - Just a minor fix in job definition to follow a new pattern.
Just updating that won't require us to rerun the kuttl job.

Thanks!

**Comment by bshewale** [comment]
> Hi @bshewale - Just a minor fix in job definition to follow a new pattern. Just updating that won't require us to rerun the kuttl job.
> 
> Thanks!

@viroel thanks for the review! :)

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 4]
```diff
@@ -49,7 +111,7 @@ presubmits:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )images,?($|\s.*)
   - agent: kubernetes
-    always_run: false
+    always_run: true
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:114]: weird, you know why those were also updated?
should be as before.

Comment (bshewale) [reply ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml]: @viroel hey i just resolved the merge conflict and ran make update after that i am seeing these changes.

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 5]
```diff
@@ -6,14 +6,76 @@ presubmits:
...
+          secretName: hive-hive-credentials
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )designate-operator-build-deploy-kuttl,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:67]: set to false

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 6]
```diff
@@ -6,14 +6,76 @@ presubmits:
...
+    - ^main-
+    cluster: build01
     context: ci/prow/images
     decorate: true
     labels:
       ci.openshift.io/generator: prowgen
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openstack-k8s-operators-designate-operator-main-images
     rerun_command: /test images
-    skip_if_only_changed: ^renovate.json|^README.md
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:16]: revert this one

File: ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml [hunk 7]
```diff
@@ -6,14 +6,76 @@ presubmits:
     - ^main$
     - ^main-
     cluster: build01
+    context: ci/prow/designate-operator-build-deploy-kuttl
+    decorate: true
+    labels:
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy-kuttl
+    optional: true
```
Comment (viroel) [inline ci-operator/jobs/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main-presubmits.yaml:15]: lets set optional to `false`
and make this job required to merge

File: ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml [hunk 8]
```diff
@@ -29,6 +66,19 @@ tests:
...
     from: openstack-operator-ci-pre-commit
+- as: designate-operator-build-deploy-kuttl
+  cluster_claim:
+    architecture: amd64
+    cloud: vexxhost
+    owner: openstack-k8s-operators
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.12"
+  optional: true
```
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:77]: optional: `false`

**Comment by viroel** [comment]
/pj-rehearse ack

there was no change in kuttl job since last run. We are just fixing always_run and optional parameters here

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-precommit-check

**Comment by viroel** [comment]
/pj-rehearse reject

until we fix precommit

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-precommit-check

**Comment by viroel** [comment]
/lgtm

thanks @bshewale for working on this

**Pull request merged**